src/main/java/com/google/sps/servlets/MealServlet.java

```diff
@@ -105,29 +116,56 @@ private void getMealById(HttpServletRequest request, HttpServletResponse respons
             response.sendError(HttpServletResponse.SC_NOT_FOUND);
             return;
         }
-
-        Meal meal = dishes.get(id);
-
-        if (meal != null) {
-            String gson = new Gson().toJson(meal);
-            response.setContentType("application/json");
-            response.getWriter().println(gson);
-            return;
+        Filter propertyFilter = new FilterPredicate("id", FilterOperator.EQUAL, id);
+        Query query = new Query("Meal").setFilter(propertyFilter);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+        ArrayList<Meal> meals = getDataFromDatastore(results); 
+        
+        if (meals.size() > 1) {
+            response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
```

Comment:
Body of this function is missing a `return` statement, please add it, otherwise the block setting `SC_NOT_FOUND` will also apply.